# 02 - Experimentos Aleatorios


## El Estándar Dorado

En el capítulo anterior, vimos por qué y cómo la _asociación_ es diferente de la _causalidad_. También, vimos lo que se requiere para que la asociación sea causalidad.

$
E[Y|T=1] - E[Y|T=0] = \underbrace{E[Y_1 - Y_0|T=1]}_{ATT} + \underbrace{\{ E[Y_0|T=1] - E[Y_0|T=0] \}}_{BIAS}
$


En resumen, la _asociación_ se convierte en _causalidad_ si no hay sesgo. No habrá sesgo si $E[Y_0|T=0]=E[Y_0|T=1]$. En palabras, la _asociación_ será _causalidad_ si el grupo control y el tratado son iguales o comparables, excepto por su tratamiento. O, en palabras más técnicas, cuando el resultado de los no tratados es igual al resultado contrafactual de los tratados. Recuerde que este resultado contrafactual es el resultado del grupo tratado si no hubiera recibido el tratamiento.

Creo que hicimos un buen trabajo al explicar cómo hacer que la _asociación_ sea igual a la _causalidad_ en términos matemáticos. Pero eso es sólo en teoría. Ahora, miremos la primera herramienta que tenemos para hacer desaparecer el sesgo: **Experimentos aleatorios**. Los experimentos aleatorios asignan aleatoriamente a los individuos de una población a un grupo de tratamiento o de control. La proporción que recibe el tratamiento no tiene que ser del 50%. Podrías tener un experimento en el que solo el 10% de sus muestras reciban el tratamiento.

La aleatorización elimina el sesgo al hacer que los resultados potenciales sean independientes del tratamiento.

$
(Y_0, Y_1) \perp\!\!\!\perp T
$

Esto puede ser confuso al principio (lo fue para mí). Pero no te preocupes, mi valiente y fiel compañero, te lo explicaré más a fondo. Si el resultado es independiente del tratamiento, ¿no implica esto también que el tratamiento no tiene efecto? ¡Bueno, sí! Pero ten en cuenta que no estoy hablando de los resultados. Más bien, estoy hablando de los resultados **potenciales**. El resultado potencial es cómo **habría sido** el resultado bajo tratamiento ($Y_1$) o bajo control ($Y_0$). En los ensayos aleatorios, **no** queremos que el resultado sea **independiente** del tratamiento, ya que pensamos que **el tratamiento causa** el resultado. Pero queremos que los resultados **potenciales** sean independientes del tratamiento. 

![img](../../causal-inference-for-the-brave-and-true/data/img/rct/indep.png)

Decir que los resultados potenciales son independientes del tratamiento es decir que, en la expectativa, serían los mismos en el grupo de tratamiento o en el de control. En términos más simples, significa que los grupos de tratamiento y control son comparables. O que conocer la asignación del tratamiento no me da ninguna información sobre cómo fue el resultado previo al tratamiento. En consecuencia, $(Y_0, Y_1)\perp T$ significa que el tratamiento es lo único que genera una diferencia entre el resultado en el grupo tratado y en el de control. Para ver esto, nótese que la independencia implica precisamente que

$
E[Y_0|T=0]=E[Y_0|T=1]=E[Y_0]
$

Lo cual, como hemos visto, hace que

$
E[Y|T=1] - E[Y|T=0] = E[Y_1 - Y_0]=ATE
$

Entonces, la aleatorización nos brinda una forma de usar una diferencia simple en las medias entre el tratamiento y el control y llamarlo el efecto del tratamiento.


## En Una Escuela Muy, Muy Lejana

En 2020, la pandemia de coronavirus obligó a las empresas a adaptarse al distanciamiento social. Los servicios de entrega se generalizaron y las grandes corporaciones cambiaron a una estrategia de trabajo remoto. Con las escuelas, no fue diferente. Muchos comenzaron su propio repositorio en línea de clases. 

A meses de la crisis, muchos se preguntan si se podrá mantener los cambios introducidos. No hay duda de que el aprendizaje en línea tiene sus beneficios. Es más barato, ya que puede ahorrar en bienes raíces y transporte. También puede ser más digital, aprovechando contenido de clase mundial de todo el mundo, no solo de un conjunto fijo de profesores. A pesar de todo eso, todavía tenemos que responder si el aprendizaje en línea tiene un impacto negativo o positivo en el rendimiento académico del estudiante.

Una forma de responder a esto es tomar a los estudiantes de escuelas que imparten principalmente clases en línea y compararlos con estudiantes de escuelas que imparten conferencias en aulas tradicionales. Como ya sabemos, este no es el mejor enfoque. Podría ser que las escuelas en línea atraigan solo a los estudiantes bien disciplinados que obtienen mejores resultados que el promedio, incluso si la clase fuera presencial. En este caso, tendríamos un sesgo positivo, donde los tratados son académicamente mejores que los no tratados: $E[Y_0|T=1] > E[Y_0|T=0]$.

Por otro lado, podría ser que las clases en línea sean más baratas y estén compuestas principalmente por estudiantes menos ricos, que quizás tengan que trabajar además de estudiar. En este caso, a estos alumnos les iría peor que a los de las escuelas presenciales aunque tomaran clases presenciales. Si este fuera el caso, tendríamos un sesgo en la otra dirección, donde los tratados son peores académicamente que los no tratados: $E[Y_0|T=1] < E[Y_0|T=0]$. 

Entonces, aunque podríamos hacer comparaciones simples, no sería convincente. De una forma u otra, nunca podríamos estar seguros de que no hubiera algún sesgo al acecho y enmascarando nuestro efecto causal.

![img](../../causal-inference-for-the-brave-and-true/data/img/rct/lurking_bias.png)

Para resolver eso, necesitamos hacer que el grupo tratado y el no tratado sean comparables $E[Y_0|T=1] = E[Y_0|T=0]$. Una forma de forzar esto es asignando aleatoriamente las clases en línea y presenciales a los estudiantes. Si lográramos hacer eso, el tratamiento y los no tratados serían, en promedio, iguales, excepto por el tratamiento que reciben. 

Afortunadamente, algunos economistas lo han hecho por nosotros. Han asignado clases aleatorias para que a algunos estudiantes las tomen presencialment, otros solo lecciones en línea y un tercer grupo para tener un formato combinado de clases en línea y presenciales. Recolectaron datos en un examen estándar al final del semestre.

Así es como se ven los datos:

In [1]:
import pandas as pd
import numpy as np

data = pd.read_csv("../../causal-inference-for-the-brave-and-true/data/online_classroom.csv")
print(data.shape)
data.head()

(323, 10)


gender  asian  black  hawaiian  hispanic  unknown  white  format_ol  \
0       0    0.0    0.0       0.0       0.0      0.0    1.0          0   
1       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
2       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
3       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
4       1    0.0    0.0       0.0       0.0      0.0    1.0          1   

   format_blended  falsexam  
0             0.0  63.29997  
1             0.0  79.96000  
2             1.0  83.37000  
3             1.0  90.01994  
4             0.0  83.30000

Podemos ver que tenemos 323 muestras. No es exactamente _Big Data_, pero es algo con lo que podemos trabajar. Para estimar el efecto causal, simplemente podemos calcular la puntuación media de cada uno de los grupos de tratamiento.

In [4]:
(data
 .assign(class_format = np.select(
     [data["format_ol"].astype(bool), data["format_blended"].astype(bool)],
     ["online", "blended"],
     default="face_to_face"
 ))
 .groupby(["class_format"])
 .mean())

gender     asian     black  hawaiian  hispanic   unknown  \
class_format                                                               
blended       0.550459  0.217949  0.102564  0.025641  0.012821  0.012821   
face_to_face  0.633333  0.202020  0.070707  0.000000  0.010101  0.000000   
online        0.542553  0.228571  0.028571  0.014286  0.028571  0.000000   

                 white  format_ol  format_blended   falsexam  
class_format                                                  
blended       0.628205        0.0             1.0  77.093731  
face_to_face  0.717172        0.0             0.0  78.547485  
online        0.700000        1.0             0.0  73.635263

Sí. Es así de simple. Podemos ver que las clases presenciales arrojan un puntaje promedio de 78.54, mientras que los cursos en línea arrojan un puntaje promedio de 73.63. No tan buenas noticias para los defensores del aprendizaje en línea. El $ATE$ para una clase en línea es, por lo tanto, -4,91. Esto significa que **las clases en línea hacen que los estudiantes rindan alrededor de 5 puntos menos**, en promedio. Eso es todo. No debe preocuparse de que los cursos en línea puedan tener estudiantes más pobres que no pueden pagar las clases presenciales o, de hecho, no debe preocuparse de que los estudiantes de los diferentes tratamientos sean diferentes de alguna manera. aparte del trato que recibieron. Por diseño, el experimento aleatorio está hecho para eliminar esas diferencias. 

Por esta razón, una buena verificación para ver si la asignación al azar se realizó correctamente (o si estás viendo los datos correctos) es revisar si los tratados son iguales a los no tratados en las variables de pre-tratamiento. Nuestros datos tienen información sobre el género y el origen étnico para ver si son similares entre los grupos. Podemos decir que se ven bastante similares para las variables "género", "asiático", "hispano" y "blanco". La variable `black`, sin embargo, parece un poco diferente. Esto llama la atención sobre lo que sucede con un conjunto de datos pequeño. Incluso bajo la aleatorización, podría ser que, por casualidad, un grupo sea diferente de otro. En muestras grandes, esta diferencia tiende a desaparecer.

## El Experimento Ideal

Los experimentos aleatorios o ensayos controlados aleatorios (_RCT_ por sus siglas en inglés) son la forma más confiable de obtener efectos causales. Es una técnica sencilla y absurdamente convincente. Es tan poderoso que la mayoría de los países lo exigen como requisito para demostrar la eficacia de los nuevos medicamentos. Para hacer una terrible analogía, puedes pensar en RCT como _Aang_, de _Avatar: The Last Airbender_, mientras que otras técnicas se parecen más a _Sokka_. _Sokka_ es genial y puede hacer algunos trucos ingeniosos aquí y allá, pero _Aang_ puede doblar los cuatro elementos y conectarse con el mundo espiritual. Piénsalo de esta manera, si pudiéramos, _RCT_ sería todo lo que haríamos para descubrir la causalidad. Un _RCT_ bien diseñado es el sueño de cualquier científico.

![img](../../causal-inference-for-the-brave-and-true/data/img/rct/science_dream.png)

Desafortunadamente, tienden a ser muy costosos o simplemente poco éticos. A veces, simplemente no podemos controlar el mecanismo de asignación. Imagínese como un médico tratando de estimar el efecto de fumar durante el embarazo sobre el peso del bebé al nacer. No se puede simplemente obligar a una porción aleatoria de madres a fumar durante el embarazo. O, digamos que trabaja para un gran banco y necesita estimar el impacto de la línea de crédito en la rotación de clientes. Sería demasiado costoso otorgar líneas de crédito aleatorias a sus clientes. O que quieras entender el impacto de aumentar el salario mínimo en el desempleo. No se puede simplemente asignar países para tener uno u otro salario mínimo.

Más adelante veremos cómo reducir el costo de la aleatorización mediante el uso de la aleatorización condicional, pero no hay nada que podamos hacer con respecto a los experimentos poco éticos o inviables. Aún así, cada vez que tratamos con cuestiones causales, vale la pena pensar en el **experimento ideal**. Pregúntate siempre, si pudieras, **¿cuál sería el experimento perfecto que harías para descubrir este efecto causal?** Esto tiende a arrojar algo de luz sobre la forma en que podemos descubrir el efecto causal incluso sin el experimento ideal.


## El Mecanismo de Asignación

En un experimento aleatorio, el mecanismo que asigna unidades a un tratamiento u otro es, bueno, aleatorio. Como veremos más adelante, todas las técnicas de inferencia causal intentarán de alguna manera identificar los mecanismos de asignación de los tratamientos. Cuando sepamos con certeza cómo se comporta este mecanismo, la inferencia causal será mucho más segura, incluso si el mecanismo de asignación no es aleatorio.

Desafortunadamente, el mecanismo de asignación no se puede descubrir simplemente mirando los datos. Por ejemplo, si tiene un conjunto de datos donde la educación superior se correlaciona con la riqueza, no puede saber con certeza cuál fue causado por un factor u otro con solo mirar los datos. Tendrás que usar tu conocimiento sobre cómo funciona el mundo para argumentar a favor de un mecanismo de asignación plausible: ¿es cierto que las escuelas educan a las personas, las hacen más productivas y las conducen a trabajos mejor pagados? O, si eres pesimista acerca de la educación, puedes decir que las escuelas no hacen nada para aumentar la productividad, y esto es solo una correlación espuria porque solo las familias ricas pueden permitirse que un niño obtenga un título superior.

En cuestiones causales, por lo general, podemos argumentar en ambos sentidos: que $X$ causa $Y$, o que es una tercera variable $Z$ la que causa tanto $X$ como $Y$, y por lo tanto la correlación $X$ e $Y$ es simplemente espuria. Por esta razón, conocer el mecanismo de asignación conduce a una respuesta causal mucho más convincente. Esto es también lo que hace que la inferencia causal sea tan emocionante. Si bien el Aprendizaje Automático (_Machine Learning_) aplicado generalmente solo presiona algunos botones en el orden correcto, la inferencia causal aplicada requiere que piense seriamente en el mecanismo que genera esos datos. 


## Ideas Clave

Observamos cómo los experimentos aleatorios son la forma más simple y efectiva de descubrir el impacto causal. Esto se logra haciendo que los grupos de tratamiento y control sean comparables. Desafortunadamente, no podemos hacer experimentos aleatorios todo el tiempo, pero aun así es útil pensar cuál es el experimento ideal que haríamos si pudiéramos.

Alguien familiarizado con la Estadística podría estar protestando en este momento porque no observé la varianza de mi estimación del efecto causal. ¿Cómo puedo saber que una disminución de 4,91 puntos no se debe al azar? En otras palabras, ¿cómo puedo saber si la diferencia es estadísticamente significativa? Y ellos estarían en lo correcto. No te preocupes. Pretendo revisar algunos conceptos estadísticos a continuación. 


## Referencias

Me gusta pensar en este libro como un tributo a Joshua Angrist, Alberto Abadie y Christopher Walters por su increíble clase de Econometría. La mayoría de las ideas aquí están tomadas de sus clases en la Asociación Económica Americana. Verlos es lo que me mantiene cuerdo durante este difícil año de 2020.
* [Cross-Section Econometrics](https://www.aeaweb.org/conference/cont-ed/2017-webcasts)
* [Mastering Mostly Harmless Econometrics](https://www.aeaweb.org/conference/cont-ed/2020-webcasts)

También me gustaría hacer referencia a los increíbles libros de Angrist. Me han demostrado que la Econometría, o 'Metria como la llaman, no solo es extremadamente útil sino también profundamente divertida.

* [Mostly Harmless Econometrics](https://www.mostlyharmlesseconometrics.com/)
* [Mastering 'Metrics](https://www.masteringmetrics.com/)

Mi referencia final es el libro de Miguel Hernan y Jamie Robins. Ha sido mi compañero de confianza en las preguntas causales más espinosas que tuve que responder.

* [Causal Inference Book](https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/)

La analogía de la cerveza fue tomada de [Stock Series](https://jlcollinsnh.com/2012/04/15/stocks-part-1-theres-a-major-market-crash-coming-and-dr-lo-cant-save-you/), por JL Colins. Esta es una lectura obligada para todos aquellos que quieran aprender a invertir su dinero de manera productiva.

![img](../../causal-inference-for-the-brave-and-true/data/img/poetry.png)

## Contribuye

_Inferencia Causal para los Valientes y Verdaderos_ es un material de código abierto sobre Inferencia Causal, la Estadística de la Ciencia. Utiliza solo software libre, basado en Python. Su objetivo es ser accesible monetaria e intelectualmente.
Si encuentras este libro valioso y quieres apoyarlo, por favor visita [Patreon](https://www.patreon.com/causal_inference_for_the_brave_and_true). Si no estás listo para contribuir financieramente, también puedes ayudar corrigiendo errores tipográficos, sugiriendo ediciones o dando tu opinión sobre pasajes que no entendiste. Sólo tienes que ir al repositorio del libro y [abrir un _issue_](https://github.com/matheusfacure/python-causality-handbook/issues). Finalmente, si te gustó este contenido, compártelo con otras personas a quienes les pueda resultar útil y dale una [estrella en GitHub](https://github.com/matheusfacure/python-causality-handbook/stargazers).